deploy/charts/templates/deployment.yaml

```diff
@@ -42,3 +42,15 @@ spec:
               name: {{ .Values.configmap.name }}
         resources:
           {{- toYaml .Values.resources | nindent 12 }}
+        readinessProbe:
+          httpGet:
+            path: /healthcheck
```

Comment:
I'm not 100% aware of what the Terminus healtcheck actually does, so I'll just provide some [generic feedback](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) here.

Readiness probes are used to determine when a pod is ready to accept traffic (meaning, capable of actually serving the request in full). It's checked throughout the life of the pod (so that pods could temporarily stop accepting traffic without needing a restart), and is allowed to fail `failureThreshold` (default 3) times in a row before being restarted. Generally you want to set your initial delay conservatively. Readiness probes often check dependencies, so you might want to extend the `periodSeconds` timeout longer if a dependency check might take longer at times due to temporary latency.

Liveness probes are used to determine whether to restart a pod after everything starts. Generally the probe determines whether the pod can be considered responsive, and has a similar `failureThreshold` mechanism as readiness probes. Often liveness probes don't check dependencies, instead preferring to have a lightweight and reliable check. Again, we may want to consider setting delays and timeouts conservatively to avoid excessive restarts if something is temporarily off.

The dependencies of the app may not imply a significant difference between "service is ready to accept traffic" and "service is alive" so the same health check may be entirely appropriate.

Reply:
I've def read through the docs on this and we could consider adding more nuance at some point.  For now I decided to just reuse a single endpoint that represents a minimal request/response cycle. 

The terminus lib allows us to create our own methods (aka. complexity) for these checks. However, the recurrence of these checks in such short succession means we need to make sure they are super efficient.

As for what either of these mean in the Ui app, we are generally "ready" when we can serve requests. That said, a couple of places we could get more diligent about would be to check if the graphql schema has been fetched and  check to verify if the cache connection is active. Now, what value do we get from these checks? If we're checking/hitting connection to cache every 5-10 seconds? If the cache is not available for 1 instance it's likely not available for any of them...

The settings here are conservative based on observation in start up and readiness of the pods.